controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -229,6 +232,49 @@ func (in *KubeadmControlPlane) validateCommon() (allErrs field.ErrorList) {
 	return allErrs
 }
 
+func (in *KubeadmControlPlane) validateEtcd(prev *KubeadmControlPlane) (allErrs field.ErrorList) {
+	if in.Spec.KubeadmConfigSpec.InitConfiguration.Etcd.External != nil && prev.Spec.KubeadmConfigSpec.InitConfiguration.Etcd.Local != nil {
+		allErrs = append(
+			allErrs,
+			field.Forbidden(
+				field.NewPath("spec", "kubeadmConfigSpec", "clusterConfiguration", "etcd", "external"),
+				"cannot have both local and external etcd at the same time",
+			),
+		)
+	}
+
+	if in.Spec.KubeadmConfigSpec.InitConfiguration.Etcd.Local != nil && prev.Spec.KubeadmConfigSpec.InitConfiguration.Etcd.External != nil {
+		allErrs = append(
+			allErrs,
+			field.Forbidden(
+				field.NewPath("spec", "kubeadmConfigSpec", "clusterConfiguration", "etcd", "external"),
+				"cannot have both local and external etcd at the same time",
+			),
+		)
+	}
+
+	if in.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd.External != nil && in.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd.Local != nil {
```

Comment:
This shouldn't reference the ClusterConfiguration, please see: https://github.com/kubernetes-sigs/cluster-api/pull/2605/files

Reply:
ok, didn't see that other PR :D